core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -58,7 +61,16 @@ def promotion
       end
 
       def successful?
-        success.present? && error.blank?
+        success.present? && errors.empty?
+      end
+
+      def error
+        Spree::Deprecation.warn "#error is deprecated. Please start using #errors."
+        errors.full_messages.first
+      end
+
+      def status_code
+        errors.any? ? errors.details[:base].first[:error_code] : @success_status_code
```

Comment:
Wondering if we need to keep the `status_code` at the handler level for some specific reason or we can move the `status_code` attribute into each error and deprecate this as well. Thoughts?

Reply:
status_code also contains codes for successful action, I wasn't sure if it is used by someone or it can be useful in some applications, hence I kept it. Doesn't seem to be used currently though.